File: Discord Webhooks/DiscordApp.lua
local HttpService = game:GetService("HttpService")

local DiscordApp = {}
DiscordApp.__index = DiscordApp

function DiscordApp.PostAsync(self, config)
	local response = HttpService:RequestAsync({
		Url = self.ApplicationURL,
		Method = "POST",
		Headers = {
			["Content-Type"] = "application/json"
		},
		Body = HttpService:JSONEncode(config)
    })
    print(response.StatusCode)
	return nil
end

function DiscordApp.Register(id: string, token: string)
	return setmetatable({
		ApplicationURL = "https://discordapp.com/api/webhooks/" .. id .. "/" .. token
	}, DiscordApp)
end

return DiscordApp

File: Discord Webhooks/Typescript/DiscordApp.ts
import { HttpService } from "@rbxts/services";

export class DiscordApp {

    id: string;
    token: string;

    constructor(id: string, token: string) {
        this.id = id;
        this.token = token;
    }

    PostAsync(body: unknown): void {
        return HttpService.RequestAsync({
            Url: `https://discordapp.com/api/webhooks/${this.id}/${this.token}`,
            Method: "POST",
            Headers: { "Content-Type": "application/json" },
            Body: HttpService.JSONEncode(body)
        });
    }

}